src/saml.ts

```diff
@@ -40,14 +40,38 @@ import {
   validateSignature,
   xpath,
 } from "./xml";
-import { certToPEM, generateUniqueId, keyToPEM } from "./crypto";
+import { keyInfoToPem, generateUniqueId, PemLabel } from "./crypto";
 import { dateStringToTimestamp, generateInstant } from "./datetime";
 import { signAuthnRequestPost } from "./saml-post-signing";
 import { generateServiceProviderMetadata } from "./metadata";
 
 const inflateRawAsync = util.promisify(zlib.inflateRaw);
 const deflateRawAsync = util.promisify(zlib.deflateRaw);
 
+export const resolveAndParseKeyInfosToPem = async ({
+  cert,
+}: Pick<SamlOptions, "cert">): Promise<string[]> => {
+  const keyInfosToHandle: Array<string | string[]> = [];
+  const pemFiles: string[] = [];
+  if (typeof cert === "function") {
+    await util
+      .promisify(cert as CertCallback)()
+      .then((certs) => {
+        assertRequired(certs, "callback didn't return cert");
+
+        keyInfosToHandle.push(certs);
```

Comment:
You might have to use a spread operator here as we don't want to have nested arrays.